### ISSUE TITLE
Raster footprint generation

### ISSUE THREAD

**Issue by drnextgis** [developmentseed/rio-stac#53]
`rio-stac` creates a geometry property using a raster extent. I found [this](https://www.element84.com/blog/the-stactools-raster-footprint-utility) article explaining how to generate convex hulls of valid raster data for use in STAC Item geometries to be very useful. I believe it would be beneficial to have this option incorporated into `rio-stac`.

**Comment by vincentsarago** [comment]
thanks for sharing @drnextgis 

I believe I've already implemented some of this in https://github.com/developmentseed/rio-stac/blob/main/rio_stac/stac.py#L53-L73 but I couldn't (or didn't want) to go further down the `optimization` because:
- I didn't wanted to add shapely dependency
- I didn't want to `read` the data (at least for the geometry part, we read the data for the statistics but we usually use overview)

**Comment by drnextgis** [comment]
Fair enough. What would you recommend for a user who loves `rio-stac` and desires to have an accurate footprint as the geometry of an item? :blush:

**Comment by vincentsarago** [comment]
The more I think about it the more I think that maybe we could add optional dependency (shapely) to allow this! 

Like if the user request `precise` geometry we check first if shapely is available 🤷
